### PR TITLE
Broken reader ut

### PR DESCRIPTION
**Pull request by nadolskit**
Screenshot 2024-09-27 at 9 04 54 AM

seems consistent now

### CODE REVIEW

File: tests/test_paperqa.py [hunk 1]
```diff
@@ -814,6 +814,10 @@ def test_pdf_reader_w_no_match_doc_details(stub_data_dir: Path) -> None:
     )
 
 
+# default: ['method', 'scheme', 'host', 'port', 'path', 'query']
```
Comment (jamesbraza) [inline tests/test_paperqa.py:817]:
```suggestion
# vcr.VCR default: ('method', 'scheme', 'host', 'port', 'path', 'query')
```

File: tests/test_paperqa.py [hunk 2]
```diff
@@ -814,6 +814,10 @@ def test_pdf_reader_w_no_match_doc_details(stub_data_dir: Path) -> None:
...
+# this is the default list + body
+# this ensures vcr distinguishes requests with a different body
```
Comment (jamesbraza) [inline tests/test_paperqa.py:819]: I am tripping up on "distinguishes". Do you mean this cassette will:
- Generate a new one if the body changes
- Not generate a new one if the body changes

Comment (nadolskit) [reply tests/test_paperqa.py:819]: Maybe I can word this better, vcr uses these properties to distinguish how requests are different.
If you look through the cassette, many of the stored requests are identical with the exception of the body (identical method, URI, headers, etc)

for example the calls to https://api.openai.com/v1/embeddings and /v1/chat/completion.

Adding body won't create new cassettes; it just helps vcr correctly match the right request with its response.

File: tests/test_paperqa.py [hunk 3]
```diff
@@ -814,6 +814,10 @@ def test_pdf_reader_w_no_match_doc_details(stub_data_dir: Path) -> None:
     )
 
 
+# vcr.VCR default: ('method', 'scheme', 'host', 'port', 'path', 'query')
+# this is the default list + body
+# Adding body won't create new cassettes; it just helps vcr correctly match the right request with its response.
+@pytest.mark.vcr(match_on=["method", "scheme", "host", "port", "path", "query", "body"])
```
Comment (jamesbraza) [inline tests/test_paperqa.py:820]:
```suggestion
# SEE: https://github.com/kevin1024/vcrpy/blob/v6.0.1/vcr/config.py#L43
VCR_DEFAULT_MATCH_ON = ('method', 'scheme', 'host', 'port', 'path', 'query')


# Adding body won't create new cassettes; it just helps vcr correctly match the right request with its response.
@pytest.mark.vcr(match_on=[*VCR_DEFAULT_MATCH_ON, "body"])
```

**Review by jamesbraza — APPROVED** [review]
Nice work here, thank you so much!

**Pull request merged**